controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +590,109 @@ func (r *OVNDBClusterReconciler) reconcileServices(
+			if test_err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", test_err))
+			}
+			err := helper.GetClient().Delete(ctx, dnsRecord)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error: %v", err))
+			}
+
 		}
 	}
+
+	Log.Info("DNS Starts here")
+
+	serviceName = ovndbcluster.ServiceNameSB
+	if instance.Spec.DBType == v1beta1.NBDBType {
+		serviceName = ovndbcluster.ServiceNameNB
+	}
+	serviceLabels = map[string]string{
+		common.AppSelector: serviceName,
+	}
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	if len(podList.Items) > int(*(instance.Spec.Replicas)) {
+
+	}
+	for i, ovnPod := range podList.Items {
+		if i >= int(*(instance.Spec.Replicas)) {
+			Log.Info(fmt.Sprintf("In iteration %v is OOO, exit", i))
+			break
+		}
+		Log.Info(fmt.Sprintf("Using Pod: %v", ovnPod.Name))
+		var dnsName string
+		var dnsIP string
+		var hostnames []string
+		dnsName = "dns-" + ovnPod.Name
+		// Get Hostname
+		svc, _ := service.GetServiceWithName(
+			ctx,
+			helper,
+			ovnPod.Name,
+			ovnPod.Namespace,
+		)
+		if err != nil {
+			return ctrl.Result{}, err
+		}
+		hostname := svc.ObjectMeta.Annotations[infranetworkv1.AnnotationHostnameKey]
+		hostnames = append(hostnames, hostname)
+
+		// Get IP
+		net_stat, _ := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+		for _, v := range net_stat {
+			if v.Interface == instance.Spec.NetworkAttachment {
+				dnsIP = v.IPs[0]
+			}
+		}
+
+		// Create DNSRecord
+		var DNSRecords []infranetworkv1.DNSHost
+		// ovsdbserver-sb-x entry
+		DNSRecord := infranetworkv1.DNSHost{}
+		DNSRecord.IP = dnsIP
+		DNSRecord.Hostnames = hostnames
+		// ovsdbserver-sb entry
+		headless_dns_hostname := ovndbcluster.ServiceNameSB + "." + instance.Namespace + ".svc"
+		if instance.Spec.DBType == v1beta1.NBDBType {
+			headless_dns_hostname = ovndbcluster.ServiceNameNB + "." + instance.Namespace + ".svc"
+		}
+		DNSRecordCname := infranetworkv1.DNSHost{}
+		DNSRecordCname.IP = dnsIP
+		DNSRecordCname.Hostnames = append(DNSRecordCname.Hostnames, headless_dns_hostname)
+		DNSRecords = append(DNSRecords, DNSRecord)
+		DNSRecords = append(DNSRecords, DNSRecordCname)
+
+		// Create DNSData
+		DNSData := &infranetworkv1.DNSData{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      dnsName,
+				Namespace: ovnPod.Namespace,
+			},
+		}
+		_, err := controllerutil.CreateOrPatch(ctx, helper.GetClient(), DNSData, func() error {
+			DNSData.Spec.Hosts = DNSRecords
+			DNSData.Spec.DNSDataLabelSelectorValue = "dnsdata"
+			err := controllerutil.SetControllerReference(helper.GetBeforeObject(), DNSData, helper.GetScheme())
+			if err != nil {
+				Log.Info(fmt.Sprintf("Inside CreateOrPatch error: %v", err))
+			}
+			return err
+		})
+		if err != nil {
+			Log.Info(fmt.Sprintf("Outside CreateOrPatch error: %v", err))
+		}
+
+	}
+
 	Log.Info("Reconciled OVN DB Cluster Service successfully")
```

Comment:
Unrelated to the change, but now i wonder if we really need NetworkAttachments for some purpose, i.e we could have used just 1 service + service overrides as 1 entrypoint as here too single dns name is used and load balance to all the ips.

Reply:
this was discarded as an option as it will lead to bottleneck on service.